src/marqo/s2_inference/s2_inference.py

```diff
@@ -188,7 +188,7 @@ def _validate_model_properties(model_name: str, model_properties: dict) -> dict:
         """checks model dict to see if all required keys are present
         """
         if model_properties.get("type", None) in (None, "sbert"):
```

Comment:
I would separate the if statements determining required_keys from the verification
```python
required_keys = []
if model_properties.get("type", None) in (None, "sbert"):
    required_keys = ["dimensions", "name"]
elif model_properties.get("type", None) in ("hf"):
    required_keys = ["dimensions"]
    
for key in required_keys:
    if key not in model_properties:
        raise InvalidModelPropertiesError(
            f"model_properties has missing key '{key}'."
            f"please update your model properties with required key `{key}`"
            f"check `https://docs.marqo.ai/0.0.12/Models-Reference/dense_retrieval/` for more info."
        )


```

Reply:
updated